### PR TITLE
[MM-51855] fixed status code for /channel

### PR DESCRIPTION
**Pull request by harshal2030**
#### Summary

Fixes the status code for /api/v4/channels for wrong json data from 403 to 400

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24291
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by harshal2030** [comment]
Hi, Is there anything else that needs a change by me, there are no reviewer yet

**Comment by hanzei** [comment]
@harshal2030 Sorry it took so long to get your PR reviewed. Taking a look right now!

File: server/channels/api4/channel.go [hunk 1]
```diff
@@ -87,6 +87,12 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// check if required fields are present
+	if channel.TeamId == "" || channel.Name == "" || channel.DisplayName == "" {
```
Comment (agnivade) [inline server/channels/api4/channel.go:91]: I think we should just verify the `Type` and `TeamId` to let the permission checks pass. The rest of the validation is taken care of by the store and app layers. Let's also add a unit test please. Thanks.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: server/channels/api4/channel.go [hunk 2]
```diff
@@ -87,6 +87,12 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		c.SetInvalidParam("channel")
```
Comment (hanzei) [inline server/channels/api4/channel.go:92]: Can we make the error more descriptive? The current output is 
```json
{
    "detailed_error": "",
    "id": "api.context.invalid_body_param.app_error",
    "message": "Invalid or missing channel in request body.",
    "request_id": "kcofxp4mwtropx683kngokqn1h",
    "status_code": 400
}
```

**Review by hanzei — CHANGES_REQUESTED** [review]

**Comment by hanzei** [comment]
@harshal2030 Please let me know if you have any questions regarding the feedback.

**Pull request closed without merging**